### DISCUSSION TITLE
How do I use the given_cookie property to find a players housing?

### DISCUSSION THREAD

**Discussion by lem6ns** [HypixelDev/PublicAPI#378] [Q&A] [answered]
I'm trying to find what housing a user has given cookies to for a housing statistics tracker. I know how the week format works (from another issue). given_cookie returns UUIDs that can be a player, but most of the time it's invalid. I would like to know how to use the UUIDs. Thanks!

**Comment by AzureAaron** [comment]
> I'm trying to find what housing a user has given cookies to for a housing statistics tracker. I know how the week format works (from another issue). given_cookie returns UUIDs that can be a player, but most of the time it's invalid. I would like to know how to use the UUIDs. Thanks!

Afaik it’s impossible with the API to relate a house’s uuid to an actual player without manually going ingame and giving them cookies. They should add an endpoint for housing really as it would be beneficial for it’s player base - #355 

I’m not sure if this works but you could see if you can use the UUID within the /visit command however I doubt that it would.

**Reply by lem6ns** [reply]
Yeah, you're right, it doesn't work using the UUID. I wonder what the invalid UUIDs are used for though. Really interesting.

I can find my own house ID by just giving cookies on an account that hasn't gave cookies this past week so it doesn't get mixed up. Maybe that's how Lemminq's housing works, checks if their housing ID is in a users given cookies. But, my house ID isn't included anywhere in my /player

**Reply by AzureAaron** [reply]
> Yeah, you're right, it doesn't work using the UUID. I wonder what the invalid UUIDs are used for though. Really interesting.
> 
> I can find my own house ID by just giving cookies on an account that hasn't gave cookies this past week so it doesn't get mixed up. Maybe that's how Lemminq's housing works, checks if their housing ID is in a users given cookies. But, my house ID isn't included anywhere in my /player

It could work like Skyblock profiles and your 1st original house is your Player UUID (e.g ed3b9805117c47f780dfd05120f1d4a3) and other houses so on are the invalid UUIDs, I have no clue. They certainly aren’t MongoDB ids which is intriguing so they have some sort of relation. My first house id may be my uuid however I would need to check it first.

Housing needs some API support tbh, it’s getting much more commonly requested and asked about. I enjoy how we’re tryna figure out how this works.

**Reply by lem6ns** [reply]
Agreed.

**Comment by 0x26e** [comment]
I believe this might be similar to Skyblock Auction UUIDs. This is that the UUIDs aren't of the player who put the auction up (thats a different stat), but rather that the Auction object itself has it's own UUID generated for it. There are most likely Housing UUIDs similar to this, where the UUID is attached to a Housing object (and thus this explains the "invalid uuids" thing that has been brought up in this thread).

**Comment by TreasonMay** [comment, marked as answer]
As answered in another question: /wtfmap gives you the UUID of the house you are currently in.
